From cfe1db3994b9c466b4bc1fc83d21d98e81a835a0 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Fri, 4 Sep 2020 18:31:35 +0900
Subject: [PATCH 29/N] Cygwin: pty: Prevent garbled output when pseudo console
 is disabled.

- If pseudo console is disabled, non-cygwin apps do not detect
  console device. In this case, the some apps may output messages
  based on the locale. In addition, some apps output UTF-8 string
  regardless of the locale setting. At least git-for-windows and
  node.js do that. Even in this case, garbled output is prevented
  with this patch in most cases because mintty uses UTF-8 by default.
---
 winsup/cygwin/fhandler_tty.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/winsup/cygwin/fhandler_tty.cc b/winsup/cygwin/fhandler_tty.cc
index 43eebc1..b9cb45b 100644
--- a/winsup/cygwin/fhandler_tty.cc
+++ b/winsup/cygwin/fhandler_tty.cc
@@ -2905,6 +2905,8 @@ fhandler_pty_slave::setup_locale (void)
 	  get_ttyp ()->term_code_page = cs_names[i].cp;
 	  break;
 	}
+  SetConsoleCP (get_ttyp ()->term_code_page);
+  SetConsoleOutputCP (get_ttyp ()->term_code_page);
 }
 
 void
-- 
2.29.2

